xgcm/grid.py

```diff
@@ -1123,6 +1123,32 @@ def diff(self, da, axis, **kwargs):
         ax = self.axes[axis]
         return ax.diff(da, **kwargs)
 
+
+    @docstrings.dedent
+    def shift(self, da, axis, to=None):
+        """
+        Shift DataArray to
+
+        Parameters
+        ----------
+        da : xarray.DataArray
+
+        axis : str
+            axis name
+
+        to : str
+            new position to shift to
+
+        Returns
+        -------
+        da_s : xarray.DataArray
+            The shifted data
+        """
+
+        # TODO: probably need boundary here
```

Comment:
Would you allow all the boundary options, like in the other operations like `.diff` etc?

The way I understand it, this would just rename the dimensions of the dataarray. If the array on e.g. the cell center has the same size as the one on the cell boundary, that's all you need. The data in the array stays untouched.

The only case where we need to intervene is if we have 'inner' and 'outer' coordinates. Then one side would potentially be cut or we would have to add values. I am just wondering if we should introduce a different keyword. On the one hand, `boundary` is familiar, but then again I am worried about the cutting. E.g. in the case of going from 'outer' to 'inner' coordinates, how would you determine which side of the array loses a slice?